File: CS/Controllers/DocumentViewerController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using DevExpress.Web.Mvc;
using DevExpress.XtraReports.Web.DocumentViewer;
using MVCxDocumentViewerWithReportServerMVC3.Models;

namespace MVCxDocumentViewerWithReportServerMVC3.Controllers {
    public class DocumentViewerController : Controller {
        public ActionResult DocumentViewer(ReportModel model) {
            return View(model);
        }

        public ActionResult DocumentViewerPartial(int reportID) {
            return PartialView("DocumentViewerPartial", new ReportModel() { ReportID = reportID });
        }

        public ActionResult DocumentViewerExportTo(int reportID) {
            MVCxDocumentViewerRemoteSourceSettings remoteSourceSettings = new MVCxDocumentViewerRemoteSourceSettings() {
                AuthenticationType = DevExpress.ReportServer.Printing.AuthenticationType.Windows,
                ReportId = reportID,
                ServerUri = "http://localhost:83/"
            };
            return DocumentViewerExtension.ExportToAsync(remoteSourceSettings, Request).Result;
        }
    }
}

File: VB/Controllers/DocumentViewerController.vb
Imports System
Imports System.Web
Imports DevExpress.XtraReports.Web.DocumentViewer

Namespace MVCxDocumentViewerWithReportServerMVC3.Controllers
    Public Class DocumentViewerController
        Inherits Controller

        Public Function DocumentViewer(ByVal model As ReportModel) As ActionResult
            Return View(model)
        End Function

        Public Function DocumentViewerPartial(ByVal reportID As Integer) As ActionResult
            Return PartialView("DocumentViewerPartial", New ReportModel() With {.ReportID = reportID})
        End Function

        Public Function DocumentViewerExportTo(ByVal reportID As Integer) As ActionResult
            Dim remoteSourceSettings As New MVCxDocumentViewerRemoteSourceSettings() With { _
                .AuthenticationType = DevExpress.ReportServer.Printing.AuthenticationType.Windows, _
                .ReportId = reportID, _
                .ServerUri = "http://localhost:83/" _
            }
            Return DocumentViewerExtension.ExportToAsync(remoteSourceSettings, Request).Result
        End Function
    End Class
End Namespace
